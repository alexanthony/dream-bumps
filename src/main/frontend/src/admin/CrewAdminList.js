import React, {PropTypes} from 'react';
import {Grid, Cell} from 'material-grid/dist';
import RaisedButton from 'material-ui/RaisedButton';
import { Link } from 'react-router';
import { connect } from 'react-redux';

import {getCrewsForSex} from '../rootReducer';

const CrewAdminList = ({crews}) => (
  <Grid>
    {crews.map(crew =>
      <Cell key={crew.id} col={4}>
        <RaisedButton label={crew.name} containerElement={<Link to={`/admin/crews/${crew.id}`} />} />
      </Cell>
    )}
  </Grid>
)

const mapStateToProps = (state, ownProps) => ({
  crews: getCrewsForSex(state)(ownProps.sex)
});

export default connect(mapStateToProps)(CrewAdminList);
