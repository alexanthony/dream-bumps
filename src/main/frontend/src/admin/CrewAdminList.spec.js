import React from 'react';
import { shallow } from 'enzyme';
import { CrewAdminList } from './CrewAdminList';
import toJson from 'enzyme-to-json';

test('Renders properly', () => {
  const crews = [{name: 'crew1', id: 1}, {name: 'Crew 2', id: 2}];
  const component = shallow(
    <CrewAdminList
      crews={crews}
    />
  );
  expect(component).toBeDefined();
  expect(toJson(component)).toMatchSnapshot()
});
