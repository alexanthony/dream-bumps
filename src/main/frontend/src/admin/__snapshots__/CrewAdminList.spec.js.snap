exports[`test Renders properly 1`] = `
<Grid>
  <Cell
    col={4}>
    <RaisedButton
      containerElement={
        <Link
          onlyActiveOnIndex={false}
          style={Object {}}
          to="/admin/crews/1" />
      }
      disabled={false}
      fullWidth={false}
      label="crew1"
      labelPosition="after"
      primary={false}
      secondary={false} />
  </Cell>
  <Cell
    col={4}>
    <RaisedButton
      containerElement={
        <Link
          onlyActiveOnIndex={false}
          style={Object {}}
          to="/admin/crews/2" />
      }
      disabled={false}
      fullWidth={false}
      label="Crew 2"
      labelPosition="after"
      primary={false}
      secondary={false} />
  </Cell>
</Grid>
`;
